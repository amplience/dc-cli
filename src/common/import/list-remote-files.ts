import fs from 'fs';

export interface RemoteFile {
  uri: string;
}

export const getRemoteFileList = (filePath: string): RemoteFile[] => {
  try {
    return JSON.parse(fs.readFileSync(filePath, 'utf-8'));
  } catch (err) {
    throw new Error(`Unable to read remote file list: ${filePath}`);
  }
};
