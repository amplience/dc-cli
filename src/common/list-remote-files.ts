import fs from 'fs';

export interface RemoteFile {
  uri: string;
}

export const getRemoteFileList = (filePath: string): RemoteFile[] => {
  try {
    console.log(filePath);
    return JSON.parse(fs.readFileSync(filePath, 'utf-8'));
  } catch (err) {
    throw new Error(`Unable to read remote file list: ${filePath}`);
  }
};

[{}];
