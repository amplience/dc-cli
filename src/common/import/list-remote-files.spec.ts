import fs from 'fs';
import { getRemoteFileList } from './list-remote-files';

jest.mock('fs');

describe('list-remote-files', () => {
  describe('getRemoteFileList', () => {
    afterEach((): void => {
      jest.resetAllMocks();
    });
    it('should return a file list containing a single file', () => {
      const remoteFiles = [{ uri: '//remote-uri' }];
      (fs.readFileSync as jest.Mock).mockReturnValue(JSON.stringify(remoteFiles));
      const result = getRemoteFileList('./remote-file-list.json');
      expect(fs.readFileSync).toHaveBeenCalledTimes(1);
      expect(fs.readFileSync).toHaveBeenCalledWith('./remote-file-list.json', 'utf-8');
      expect(result).toEqual(remoteFiles);
    });

    it('should throw an error when remote file list file contains invalid json', async () => {
      const remoteFiles = 'invalid json';
      (fs.readFileSync as jest.Mock).mockReturnValue(remoteFiles);
      expect(() => getRemoteFileList('./remote-file-list.json')).toThrowErrorMatchingSnapshot();
    });

    it('should throw an error when remote file list file does not exist', async () => {
      (fs.readFileSync as jest.Mock).mockImplementationOnce(() => {
        throw new Error('File does not exist');
      });
      expect(() => getRemoteFileList('./remote-file-list.json')).toThrowErrorMatchingSnapshot();
    });
  });
});
