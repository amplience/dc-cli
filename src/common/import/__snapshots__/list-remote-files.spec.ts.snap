// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list-remote-files getRemoteFileList should throw an error when remote file list file contains invalid json 1`] = `"Unable to read remote file list: ./remote-file-list.json"`;

exports[`list-remote-files getRemoteFileList should throw an error when remote file list file does not exist 1`] = `"Unable to read remote file list: ./remote-file-list.json"`;
